### PR TITLE
iterate through methodologies

### PR DESCRIPTION
**Pull request by thehighestprimenumber**

### CODE REVIEW

File: app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx [hunk 1]
```diff
@@ -317,11 +330,12 @@ function SubSectorPage({
             </MotionTabList>
 
             <TabPanels>
-              {isLoading ? (
+              {isLoading || !refNumber ? (
```
Comment (lemilonkh) [inline app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx:333]: This means that if the refNumber is missing, it keeps showing the loading spinner indefinitely. Better behavior might be to show a warning icon and/ or message (can be coordinated with the design team, for now a warning icon is sufficient).

File: app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx [hunk 2]
```diff
@@ -117,6 +117,19 @@ function SubSectorPage({
...
+      if (!res.ok) {
+        throw new Error("Failed to fetch data");
+      }
+      const data = await res.json();
+      setRefNumber(data.data.referenceNumber);
+    };
+
+    fetchData();
+  }, []);
+
```
Comment (lemilonkh) [inline app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx:132]: Please use the a Redux Toolkit API hook (defined in `src/services/api/ts`) to fetch API data.

File: app/src/components/Tabs/Activity/activity-tab.tsx [hunk 3]
```diff
@@ -89,23 +64,34 @@ const ActivityTab: FC<ActivityTabProps> = ({
   const [selectedMethodology, setSelectedMethodology] = useState("");
   const [isUnavailableChecked, setIsChecked] = useState<boolean>(false);
   const [hasActivityData, setHasActivityData] = useState<boolean>(false);
-  const [methodology, setMethodology] = useState<MethodologyValues>();
+  const [methodology, setMethodology] = useState<Methodology>();
+
+  function getMethodologies() {
+    const refNumberWithScope = refNumber + "." + (filteredScope.scope || 1);
+    const methodologies =
+      MANUAL_INPUT_HIERARCHY[refNumberWithScope]?.methodologies || [];
```
Comment (lemilonkh) [inline app/src/components/Tabs/Activity/activity-tab.tsx:72]: I would recommend to extract `MANUAL_INPUT_HIERARCHY[refNumberWithScope]` into a variable and doing a null check beforehand, along with maybe doing object destructuring and then using the default operator (`??`).

File: app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx (hunk 2 above)
Comment (cephaschapa) [reply app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx:132]: Also just a note you can get the required subsector data from this line of code, should be on line 82. This way we don't have multiple api calls
```
const subSectorData = sectorData?.subSectors.find(
    (subsectorItem) => subsectorItem.subsectorId === subsector,
  );
```

File: app/src/util/form-schema/index.ts [hunk 4]
```diff
@@ -1,3 +1,46 @@
 import SectorFormSchema from "./sector-form-schema.json";
+import HIERARCHY from "./manual-input-hierarchy.json";
```
Comment (cephaschapa) [inline app/src/util/form-schema/index.ts:2]: The new json file was already loaded and imported in the my recent PR. Did you base this PR on that?

File: app/src/components/Tabs/Activity/activity-tab.tsx [hunk 5]
```diff
@@ -89,23 +64,33 @@ const ActivityTab: FC<ActivityTabProps> = ({
...
+  const [methodology, setMethodology] = useState<Methodology>();
+
+  function getMethodologies() {
+    const refNumberWithScope =
+      referenceNumber + "." + (filteredScope.scope || 1);
+    const hierarchy = MANUAL_INPUT_HIERARCHY[refNumberWithScope] || {};
+    const methodologies = hierarchy?.methodologies || [];
+    const directMeasure = {
+      ...hierarchy.directMeasure,
+      id: hierarchy.directMeasure?.id || "direct-measure", // adds a fallback generic id for direct measure
```
[the remark this answers is not in the document]
Comment (thehighestprimenumber) [reply app/src/components/Tabs/Activity/activity-tab.tsx:76]: Evan told me to add a fallback for direct measure :shrug:

**Review by lemilonkh — APPROVED** [review]

**Pull request merged**